Add known issue for agent 8.6.2 about not migrating unencrypted config (#195)
elastic/ingest-docs: docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc

Settings with input blocks, such as Max agents are still effective.
====

[[known-issue-issue-2249-8.6.2]]

> .{agent} does not load fleet configuration after upgrade from unencrypted config

Reviewer: You need to specify which versions have unencrypted configuration. If you are a user how do you know this applies to you?

Reviewer: Save them from having to read the full summary, which is collapsed by default

> .{agent} does not load fleet configuration after upgrade from unencrypted config

Reviewer: ```suggestion
.{agent} does not load {fleet} configuration after upgrade from unencrypted config
```

[%collapsible]
====
*Details* +
{agent} will not load `fleet.yml` information after upgrading to a version that uses encrypted configuration (> 8.3).

> For a managed agent the symptom manifests as the inability to connect back to fleet after upgrade and the following log:

Reviewer: ```suggestion
For a managed agent the symptom manifests as the inability to connect to {fleet} after upgrade with the following log:
```

[source,shell]
----
Error: fleet configuration is invalid: empty access token
...
----
For more information refer to {agent-issue}2249[#2249].

*Impact* +
{agent} loses the ability to connect back to {fleet} after upgrade.

> Fix for this issue is available in version 8.7.0 and higher, for versions 8.3-8.6 a re-enroll is necessary for the agent to connect back to {fleet}

Reviewer: Does this actually affect 8.3-8.6? I thought it only affected 8.6?

Author: I thought I had tested 8.3 and 8.4 in the original issue but I actually didn't, I only tested 8.2 -> 8.6
I will test 8.2 -> 8.3, 8.2 -> 8.4, 8.2 ->8.5 upgrade scenario to confirm

Author: 8.2.0 -> 8.3.3, 8.2.0 ->8.4.3, 8.2.0 -> 8.5.3 all work correctly I will update the description to reflect that

Settings with input blocks, such as Max agents are still effective.
====

> [[known-issue-issue-2249-8.6.2]]

Reviewer: This can be copied to the 8.6.1 and 8.6.0 release notes which are in this same document. This is a bit annoying but let's make sure users are likely to read this regardless of the version of 8.6 they are targeting.

* 8.6.1 notes https://github.com/elastic/ingest-docs/blob/main/docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc#release-notes-8.6.1
* 8.6.0 notes https://github.com/elastic/ingest-docs/blob/main/docs/en/ingest-management/release-notes/release-notes-8.6.asciidoc#release-notes-8.6.0

> .{agent} does not load {fleet} configuration after upgrading from {agent} version 8.2 or earlier

Reviewer: ```suggestion
.{agent} does not load {fleet} configuration when upgrading to {agent} version 8.6.x from version 8.2 or earlier
```

Based on https://github.com/elastic/ingest-docs/pull/195#discussion_r1185267796 the affected versions still need to be clearer. I want to be extra clear that this only affects upgrades to 8.6.x, even though these are the 8.6 release notes.

> .{agent} does not load {fleet} configuration when upgrading to {agent} version 8.6.x from version 8.2 or earlier

Reviewer: ```suggestion
.{agent} does not load {fleet} configuration when upgrading to {agent} version 8.6.x from version 8.2.x or earlier
```

For consistency we should use 8.2.x to ensure people assume this applies to all 8.2.x version. I'll just make this comment one time but it applies in multiple places.

[%collapsible]
====
*Details* +

> {agent} will not load `fleet.yml` information after upgrading from a version up to 8.2 to 8.6.

Reviewer: ```suggestion
{agent} will not load `fleet.yml` information after upgrading to {agent} version 8.6.x from version 8.2 or earlier
```

Let's just use the same wording for versions from above.
